webapp/advantage/ua_contracts/parsers.py

```diff
@@ -86,6 +87,7 @@ def parse_entitlements(raw_entitlements: dict) -> List[Entitlement]:
 
 def parse_contract_items(raw_items: dict) -> List[ContractItem]:
     items = []
+    raw_items = raw_items or []
```

Comment:
I might be wrong, but if this is a dict, which I assume is `{}` but the alternative is an empty `[]` these are 2 different types? So shouldn't this be `raw_items or None`, since this is what you've done this in the other functions and args?

Reply:
I changed the `raw_items: dict` to `raw_items: List[Dict] = None` as it's more accurate.
Made this change for other parsers: `parse_entitlements`, `parse_subscriptions`, `parse_product_listings`.

`raw_items = raw_items or []` this line sets the `raw_items` to empty list, so that the loop on the next line won't throw a 500, because we try to iterate a None value.

Added tests for all these parsers in case the parameter received is a None value or an empty list.